tasklist/importer/src/test/java/io/camunda/tasklist/zeebeimport/ImportListenerElasticsearchIT.java

```diff
@@ -0,0 +1,290 @@
+    } catch (final PersistenceException e) {
+      // ignore
+    }
+
+    // when the job is executed
+    importJob.call();
+
+    // then
+    assertFalse(importListener.isFinishedCalled()); // empty batches do not trigger finished
+    assertFalse(importListener.isFailedCalled());
+    assertNull(importListener.getImportBatch());
+  }
+
+  @Test
+  public void testFailed() throws Exception {
+    final SearchHit mockedSearchHit = mock(SearchHit.class);
+    final String indexName = "some_name-8.7.0_";
+    when(mockedSearchHit.getIndex()).thenReturn(indexName);
+    when(mockedSearchHit.getSourceAsString()).thenReturn("{\"position\":12345,\"sequence\":67890}");
+    final ImportBatchElasticSearch importBatchElasticSearch =
+        new ImportBatchElasticSearch(
+            1, ImportValueType.PROCESS_INSTANCE, Arrays.asList(mockedSearchHit), indexName);
+    final ImportPositionEntity previousPosition =
+        new ImportPositionEntity().setAliasName("alias").setPartitionId(1).setPosition(0);
+    final ImportJob importJob =
+        beanFactory.getBean(
+            ImportJobElasticsearch.class, importBatchElasticSearch, previousPosition);
+    // mock import methods
+    try {
+      doThrow(new PersistenceException())
+          .when(elasticsearchBulkProcessor)
+          .performImport(importBatchElasticSearch);
+    } catch (final PersistenceException e) {
+      // ignore
+    }
+
+    // when the job is executed
+    importJob.call();
+
+    // then
+    assertTrue(importListener.isFailedCalled());
+    assertFalse(importListener.isFinishedCalled());
+    assertEquals(importListener.getImportBatch(), importBatchElasticSearch);
+  }
+
+  @Test
+  public void testFailedEmptyBatch() throws Exception {
+    final ImportBatchElasticSearch importBatchElasticSearch =
+        new ImportBatchElasticSearch(1, ImportValueType.PROCESS_INSTANCE, new ArrayList<>(), null);
+    final ImportPositionEntity previousPosition =
+        new ImportPositionEntity().setAliasName("alias").setPartitionId(1).setPosition(0);
+    final ImportJob importJob =
+        beanFactory.getBean(
+            ImportJobElasticsearch.class, importBatchElasticSearch, previousPosition);
+    // mock import methods
+    try {
+      doThrow(new PersistenceException())
+          .when(elasticsearchBulkProcessor)
+          .performImport(importBatchElasticSearch);
+    } catch (final PersistenceException e) {
+      // ignore
+    }
+
+    // when the job is executed
+    importJob.call();
+
+    // then
+    assertFalse(importListener.isFailedCalled()); // empty batches do not trigger failed
+    assertFalse(importListener.isFinishedCalled());
+    assertNull(importListener.getImportBatch());
+  }
+
+  @Test
+  public void testBatchSizeLimitingSplitsBatchesCorrectly() throws Exception {
+    // given a small configured batch size (e.g., 150 bytes)
+    tasklistProperties.getImporter().setMaxBatchSizeBytes(150L);
```

Comment:
unfortunately we have to do weird sizing in the tests here because we mock the hits, so the sizing is based on the mocked object rather than a realistic expected size